From b9902759e096cde101c18ca27485457cac96bd74 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Mon, 4 Mar 2024 18:09:53 +0100
Subject: [PATCH 1/2] raster.c: Use sRGB if driver is pwg/urf and has
 ColorModel option

Some driverless printers (EPSON L3160 in Fedora report) stopped working
after commit c6175a2 if `ColorModel=RGB` is passed as option. A
different CUPS color space is assigned with the fix - CUPS_CSPACE_RGB,
which results in no ICC profile being assigned into Ghostscript command
line.

Probably we can try other .icc profiles with CUPS_CSPACE_RGB (srgb.icc
does not work with RGB color space), but I tested with reporter that
using sRGB space + srgb.icc works for the printer - so the patch is to
use sRGB if the driver is URF/PWG.
---
 cupsfilters/raster.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cupsfilters/raster.c b/cupsfilters/raster.c
index e9a8ad1f..8a4af4a2 100644
--- a/cupsfilters/raster.c
+++ b/cupsfilters/raster.c
@@ -1571,7 +1571,7 @@ raster_base_header(cups_page_header_t *h,  // O - Raster header
     {
       if (*(val + 3) == '_' || *(val + 3) == '-')
 	ptr = val + 4;
-      colorspace = 1;
+      colorspace = pwg_raster ? 19 : 1;
       numcolors = 3;
     }
     else if (!strcasecmp(val, "auto"))

From bcd44ffcec6cfafcce234a2f008776e816047c12 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Thu, 7 Mar 2024 13:24:10 +0100
Subject: [PATCH 2/2] Apply the same logic to Gray

---
 cupsfilters/raster.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cupsfilters/raster.c b/cupsfilters/raster.c
index 8a4af4a2..7b62b922 100644
--- a/cupsfilters/raster.c
+++ b/cupsfilters/raster.c
@@ -1550,7 +1550,7 @@ raster_base_header(cups_page_header_t *h,  // O - Raster header
     {
       if (*(val + 4) == '_' || *(val + 4) == '-')
 	ptr = val + 5;
-      colorspace = 18;
+      colorspace = pwg_raster ? 18 : 3;
       numcolors = 1;
     }
     else if (is_color && !strncasecmp(val, "Srgb", 4))
